OverlayPlugin/PluginLoader.cs

```diff
@@ -62,36 +61,40 @@ public void InitPlugin(TabPage pluginScreenSpace, Label pluginStatusText)
                 return;
             }
 
+            pluginStatusText.Text = Resources.InitRuntime;
+
             Initialize();
         }
 
-        // AssemblyResolver でカスタムリゾルバを追加する前に PluginMain が解決されることを防ぐために、
-        // インライン展開を禁止したメソッドに処理を分離
+        // To prevent PluginMain from being resolved before we add our custom resolver in AssemblyResolver,
+        // we separate the reference to its own method and prevent inlining.
         [MethodImpl(MethodImplOptions.NoInlining)]
         private async void Initialize()
         {
-            pluginStatusText.Text = Resources.InitRuntime;
+            // Use TinyIoC's singleton since we don't use multiple IoC containers
+            // This won't cause issues with other plugins, as our TinyIoCContainer is namespaced to `RainbowMage.OverlayPlugin`
+            Container = TinyIoCContainer.Current;
 
-            var container = new TinyIoCContainer();
-            logger = new Logger();
-            container.Register(logger);
-            container.Register<ILogger>(logger);
+            TinyIoCAutoHelper.RegisterAssemblies(SanityChecker.GetAssemblies());
+
+            TinyIoCAutoHelper.AutoRegisterPreInit();
+            TinyIoCAutoHelper.AutoConstructPreInit();
+
+            logger = Container.Resolve<ILogger>();
 
             asmResolver.ExceptionOccured += (o, e) => logger.Log(LogLevel.Error, Resources.AssemblyResolverError, e.Exception);
             asmResolver.AssemblyLoaded += (o, e) => logger.Log(LogLevel.Info, Resources.AssemblyResolverLoaded, e.LoadedAssembly.FullName);
```

Comment:
If you register stuff to the IoC container before setting up the assmbly logging handlers, you'll miss most of the assembly log messages and errors (!).
The previous code only references types from `OverlayPlugin.Common` which doesn't have any references to other assemblies to make sure that no other assemblies are loaded before the logging is set up.